### PR TITLE
Fix/prevent calls to getter when default value is nil

### PR DESCRIPTION
**Pull request by arjan0307**

### CODE REVIEW

**Comment by CUnknown** [comment]
I would also like to have this performance optimalisation :+1:

**Comment by lest** [comment]
Looks good to me 👍
@arjan0307 Could you please fix the build (there is a failed RSpec matcher test)? Thanks!

**Comment by arjan0307** [comment]
@lest Done

File: spec/enumerize/integrations/rspec/matcher_spec.rb [hunk 1]
```diff
@@ -221,7 +221,7 @@ def self.name
   describe '#with_multiple' do
 
     it 'accepts when has defined the multiple' do
-      model.enumerize(:sex, :in => [:male, :female], multiple: true)
+      model.enumerize(:sex, :in => [:male, :female], multiple: true, default: :male)
```
Comment (lest) [inline spec/enumerize/integrations/rspec/matcher_spec.rb]: This test looks legit and I think that we shouldn't change it. I think that [the matcher itself](https://github.com/brainspec/enumerize/blob/master/lib/enumerize/integrations/rspec/matcher.rb) needs to be changed to work for cases when `default` is not set.

**Comment by lest** [comment]
I merged the changes to the master branch manually. Thanks!

PS. I adjusted the RSpec matcher to not rely on the fact that the default value is always assigned.

**Pull request closed without merging**